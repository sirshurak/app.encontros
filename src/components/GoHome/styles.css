#go-home {
    height: 60px;
    width: 60px;
    position: absolute;
}