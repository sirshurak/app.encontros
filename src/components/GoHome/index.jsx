import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faLongArrowAltLeft } from "@fortawesome/free-solid-svg-icons";
import AppContext from "../../contexts/AppContext";

import "./styles.css";

function GoHome() {
  return (
    <AppContext.Consumer>
      {({ changePage }) => (
        <button
          id="go-home"
          className="link"
          type="button"
          onClick={() => changePage("Home")}
        >
          <FontAwesomeIcon
            icon={faLongArrowAltLeft}
            size="2x"
            color="#3498db"
          />
        </button>
      )}
    </AppContext.Consumer>
  );
}
export default GoHome;
