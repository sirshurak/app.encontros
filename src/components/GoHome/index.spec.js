import { shallow, mount } from 'enzyme';
import GoHome from './index';
import AppContext from '../../contexts/AppContext';

describe('GoHome component', () => {
    it('should match shallow snapshop', () => {
        const component = shallow(<GoHome />)
        expect(component).toMatchSnapshot();
    });

    it('should mount and click button', () => {
        const changePage = jest.fn();
        const component = mount(
            <AppContext.Provider value={{ changePage }}>
                <GoHome />
            </AppContext.Provider>
        );
        component.find('button').simulate('click', {});
        expect(changePage).toBeCalled();
    });
});