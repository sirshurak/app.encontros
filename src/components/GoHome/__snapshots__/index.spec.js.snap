// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoHome component should match shallow snapshop 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
